pull request for test cases and downloading exercises,content and topics

gwong89 opened the pull request.

gwong89 committed:
pull request for test cases and downloading exercises,content and topics

aronasorman commented:
Merge conflict. Can you resolve it? There's instructions in this PR on how to resolve it.

gwong89 committed:
fixed merge conflict

aronasorman commented:
Still has conflcits.

gwong89 committed:
retrieve subtitles and caching complete

gwong89 committed:
merge, and pull request. deleted i18helpers

aronasorman commented on contentpacks/khanacademy.py:
```diff
@@ -79,8 +79,34 @@ def retrieve_language_resources(lang: str, version: str) -> LangpackResources:
-def retrieve_subtitles(videos: list, lang="en") -> dict:
-    return {}
+def retrieve_subtitles(videos: list, lang="en") -> list:
+    #videos => contains list of youtube ids
```
Rather than having this is a comment, make it a doctring instead.

aronasorman commented on tests/test_contentpacks.py:
```diff
@@ -19,6 +19,46 @@
+        empty_list = retrieve_subtitles(incorrect_list)
+        test_list = []
+        assert type(empty_list) == list
+        assert len(empty_list) == 0
```
You can summarize these last two asserts as `assert not empty_list`.

aronasorman commented on tests/test_contentpacks.py:
```diff
@@ -19,6 +19,46 @@
+        incorrect_list = ["aaa"]
+        empty_list = retrieve_subtitles(incorrect_list)
+        test_list = []
+        assert type(empty_list) == list
```
Use `isinstance` instead.

aronasorman commented on tests/test_contentpacks.py:
```diff
@@ -19,6 +19,46 @@
+        filled_list = retrieve_subtitles(mixed_list)
+        test_list = ["y2-uaPiyoxc"]
+        assert type(filled_list) == list
+        assert len(filled_list) == 1
```
Read up on Python's truthiness idioms and then try to remove the redundancies in your assert statements.

aronasorman commented on contentpacks/khanacademy.py:
```diff
@@ -79,8 +79,34 @@ def retrieve_language_resources(lang: str, version: str) -> LangpackResources:
+    for youtube_id in videos:
+        request_url = "https://www.amara.org/api2/partners/videos/?format=json&video_url=http://www.youtube.com/watch?v=%s" % (
+        youtube_id)
+        response =  requests.get(request_url)
```
The idiom in the requests library is to use `raise_for_status` to check if we got a nonsucessful error code: http://docs.python-requests.org/en/latest/user/quickstart/

gwong89 replied:
are you referring for line 90 or 100? 90 will always return a 200 regardless of a bad youtube id, unless the server is down then I don't see a point using it over there, unless you meant 100

aronasorman commented on contentpacks/khanacademy.py:
```diff
@@ -79,8 +79,34 @@ def retrieve_language_resources(lang: str, version: str) -> LangpackResources:
+
+            except urllib.error.HTTPError:
+                continue   
+            FILE_DIR = os.path.join(os.getcwd(), "build", "subtitles", lang)
```
Don't make variables all caps if they aren't constants.

aronasorman commented:
Merge conflict.

gwong89 committed:
merge and fixed some pr stuff
new merge for pr request
adding print statement
added force flag to retrieve_subtitles function
